torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -1647,53 +1647,108 @@ def codegen_alloc_from_pool(self, name, offset, dtype, shape, stride) -> str:
+                        writer,
+                        known_statically=self.is_statically_known_list_of_ints(
+                            stride_list
+                        ),
+                        graph=self.get_codegened_graph(),
+                    ),
+                    offset,
+                ]
+                call_str = (
                     f"auto {tmp_name} = reinterpret_tensor_wrapper({', '.join(args)});"
                 )
+                return tmp_name, call_str
+
+            def create_dtypeview_call(reinterpret_call):
+                tmp_output_name = (
+                    f"tmp_{data.get_name()}_handle_{next(self.tmp_tensor_id)}"
+                )
+                call_strs = [f"AtenTensorHandle {tmp_output_name};"]
+                dtype_name = str(dtype).split(".")[-1]
+                # @TODO how to check if it is real?
+                dtypeview_suffix = "_dtype" if not dtype.is_complex else "_as_complex"
+                get_dtype_function = f"aoti_torch_dtype_{dtype_name}"
+                device_name = "cuda" if data.layout.device.type == "cuda" else "cpu"
+                dtypeview_function = f"aoti_torch_{device_name}_view{dtypeview_suffix}"
+                call_strs.append(
+                    f"AOTI_TORCH_ERROR_CODE_CHECK({dtypeview_function}"
+                    f"({reinterpret_call}, {get_dtype_function}(), &{tmp_output_name}));"
+                )
+                tmp_RAIIAtenTensorHandle = (
+                    f"tmp_{data.get_name()}_{next(self.tmp_tensor_id)}"
+                )
+                call_strs.append(
+                    f"RAIIAtenTensorHandle {tmp_RAIIAtenTensorHandle}({tmp_output_name});"
+                )
+                return tmp_RAIIAtenTensorHandle, call_strs
 
+            if (
+                size_list == data.layout.size
+                and stride_list == data.layout.stride
+                and original_offset == data.layout.offset
+            ):
+                # pure dtypeview
+                if dtype is not None and dtype != data.dtype:
+                    tmp_output_name, tmp_call_strs = create_dtypeview_call(
+                        data.get_name()
+                    )
+                    call_strs.extend(tmp_call_strs)
+                    final_tmp_name = tmp_output_name
+                    final_tmp_name_is_RAIIAtenTensorHandle = True
+                else:
+                    return f"{data.get_name()}"
+            else:
+                # firstly create reinterpretview
+                final_tmp_name, reinterpret_call = create_reinterpret_call()
+                call_strs.append(reinterpret_call)
+
+                if dtype is not None and dtype != data.dtype:
+                    # wrap it with dtypeview
+                    final_tmp_name, tmp_call_strs = create_dtypeview_call(
+                        reinterpret_call
+                    )
+                    call_strs.extend(tmp_call_strs)
+
+            if writer is None:
```

Comment:
The earlier comments, 
```
            # Because the memory planning is done in two passes (see the implementation
            # of self.generate), the writeline behavior is different in the two passes.
```
are for this. Move them over.

Reply:
fixed in https://github.com/pytorch/pytorch/pull/128883/commits/5abff9fb70f53fd321f8ad064f973734c33d0643